packages/screens/Settings/components/UserProfileModal.tsx

```diff
@@ -0,0 +1,325 @@
+import React, { useState } from "react";
+import {
+  View,
+  Modal,
+  StyleSheet,
+  useWindowDimensions,
+  Pressable,
+} from "react-native";
+import { DragSortableView } from "react-native-drag-sort";
+import Hoverable from "react-native-hoverable";
+
+import confirmSVG from "../../../../assets/icons/confirm.svg";
+import { BrandText } from "../../../components/BrandText";
+import { SVG } from "../../../components/SVG";
+import { PrimaryButton } from "../../../components/buttons/PrimaryButton";
+import { TertiaryButton } from "../../../components/buttons/TertiaryButton";
+import {
+  neutralA3,
+  neutral33,
+  neutral00,
+  neutral17,
+  neutral55,
+} from "../../../utils/style/colors";
+import {
+  fontMedium14,
+  fontSemibold14,
+  fontSemibold28,
+} from "../../../utils/style/fonts";
+import { layout } from "../../../utils/style/layout";
+
+type toggleUserProfileProps = {
+  onClose: () => void;
+  visible: boolean;
+};
+
+const smallMobileWidth = 512;
+const mobileWidth = 768;
+
+export const UserProfileModal: React.FC<toggleUserProfileProps> = ({
+  onClose,
+  visible,
+}) => {
+  const { width, height } = useWindowDimensions();
+
+  const calculateChildrenWidth = (): number => {
+    let targetWidth = 0;
+
+    const gapWidth = layout.padding_x1;
+    const miniumWidth = 240;
+
+    if (width < (miniumWidth + gapWidth * 2) * 8)
+      targetWidth = (width - gapWidth * 2 * 7) / 7;
+    if (width < (miniumWidth + gapWidth * 2) * 7)
+      targetWidth = (width - gapWidth * 2 * 6) / 6;
+    if (width < (miniumWidth + gapWidth * 2) * 6)
+      targetWidth = (width - gapWidth * 2 * 5) / 5;
+    if (width < (miniumWidth + gapWidth * 2) * 5)
+      targetWidth = (width - gapWidth * 2 * 4) / 4;
+    if (width < (miniumWidth + gapWidth * 2) * 4)
+      targetWidth = (width - gapWidth * 2 * 3) / 3;
+    if (width < (miniumWidth + gapWidth * 2) * 3)
+      targetWidth = (width - gapWidth * 2 * 2) / 2;
+    if (width < (miniumWidth + gapWidth * 2) * 2)
+      targetWidth = (width - gapWidth * 2 * 1) / 1;
+
+    return targetWidth;
+  };
+
+  const childrenHeight = 42;
+  const childrenWidth = calculateChildrenWidth();
+  const parentWidth = width;
+  const marginChildrenVertical = layout.padding_x1;
+  const marginChildrenTop =
+    width < smallMobileWidth ? layout.padding_x1 : layout.padding_x2;
+
+  interface ProfileDataType {
+    id: number;
+    content: string;
+  }
+  const initProfileData: ProfileDataType[] = [
```

Comment:
maybe we should extract this into it's own file, what do you think @3mp8r3 
cc: @WaDadidou